widget_driver/lib/src/drivable_widget.dart

```diff
@@ -55,10 +57,32 @@ class _DriverWidgetState<Driver extends WidgetDriver> extends State<DrivableWidg
     return widget.build(context);
   }
 
+  @override
+  void didUpdateWidget(covariant DrivableWidget<Driver> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (_driverExists) {
+      widget.driverProvider.updateDriverProvidedProperties(_driver!);
+    }
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (_driverExists) {
+      // If didChangeDependencies get's called this means your driver depends on a value from the BuildContext,
+      // which the flutter framework marked as "has changed".
+      // Thus we need to rebuild the driver to retrieve the latest value.
+      _markDriverToBeRebuilt();
+    }
+  }
+
   Driver _getDriverAndSetupUpIfNeeded(BuildContext context) {
-    // The driver was already created. Nothing more to do. Just return it.
-    if (_driver != null) {
-      return _driver!;
+    if (_driverExists) {
+      if (_shouldRebuildDriver) {
+        _disposeDriver();
+      } else {
+        return _driver!;
+      }
     }
 
```

Comment:
The logic here took my brain some time to parse, due to the return in the one flow and no return in the other. In general I would always try to have only one return in a function and not have them spread out over the code of a function.
Special case are "early exits" like guard statements in Swift. 
For me it would also be ok to do this here, but not mix it with other code, so I'd suggest:
```suggestion
Driver _getDriverAndSetupUpIfNeeded(BuildContext context) {
  if (_driverExists && !_shouldRebuildDriver) {
    // The existing driver is fine. Nothing more to do. Just return it.
    return _driver;
  }
  
  _disposeDriver(); // do this for all cases - see my later comment
  // no further returns until the last line of the function.
}
```